cranelift-codegen/src/legalizer/boundary.rs

```diff
@@ -245,6 +266,176 @@ where
     call
 }
 
+fn assert_is_valid_sret_legalization(
```

Comment:
Should the function definition be guarded by a cfg guard too, to avoid dead code warnings in release mode?

Reply:
Because the call is inside a `if cfg!(...) { assert_blah(); }` it will expand to `if false { assert_blah(); }` which won't trigger dead code warnings, but will get DCE'd.

If instead of `if cfg!(...) { ... }`, we did

```rust
#[cfg(debug_assertions)]
{
    ...
}
```

then the block would expand into nothing, and there would be no caller of the assertion function, and it would trigger a dead code warning in release builds.